src/pages/_app.tsx

```diff
@@ -11,15 +11,26 @@ import {
 	TorusWalletAdapter,
 } from "@solana/wallet-adapter-wallets";
 import { config } from "config";
-import { MainMenu } from "@components/MainMenu";
-import { ClientProvider } from "@components/ClientProvider";
+import { NextPage } from "next"
+import { ClientProvider } from "@components/ClientProvider"
 
 // Use require instead of import since order matters
 require("@solana/wallet-adapter-react-ui/styles.css");
 require("../styles/antd.less");
 require("../styles/globals.css");
 
-const CredixApp: React.FC<AppProps> = ({ Component, pageProps }) => {
+export type NextPageWithLayout = NextPage & {
```

Comment:
I don't really like this way of adding a layout to a page. it is unnecessarily complex imo.
all these types here can go and just add the layout on the page itself

Reply:
I agree that it's complex, but not unnecessarily. 
If we add the layout to the component itself (as suggested in the comment below) React will re-render the entire layout on each route change. 

You can test this yourself by adding an input to the sidebar and giving it a value: when you add the layout directly and navigate to a new page, the input will be cleared; when the layout is added via `getLayout`, the input remains when navigating. 

This way of adding layouts to the application is taken from the Nextjs [docs](https://nextjs.org/docs/basic-features/layouts).
It's also well explained [in this blog post](https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/), which I think inspired the Nextjs docs.

It might be an optimization that we don't need (yet) but as it's a suggested way of doing per page layouts, I also don't see why we shouldn't do it this way.